app/models/agents/aftership_agent.rb

```diff
@@ -0,0 +1,139 @@
+require 'uri'
+
+module Agents
+  class AftershipAgent < Agent
+
+    default_schedule "every_10m"
+
+    description <<-MD
+      The Aftership agent allows you to track your shipment from aftership and emit them into events.
+
+      To be able to use the Aftership API, you need to generate an `API Key`. You need a paying plan to use their tracking feature.
+
+      You can use this agent to retrieve tracking data. You have to provide a specific `path` request and its associated option.
+ 
+      To get all trackings for your packages please enter `path` for key and `trackings` for the option.
+      To get tracking for a specific tracking number, add the extra keys `slug`, `tracking_number` and their associated values. Set `single_tracking_request` to true.
+
+      To get the last checkpoint of a package set key to `path` and option to `last_checkpoint`. Please provide `slug` and `tracking_number`. Set `last_checkpoint_request` to true.
+
+      `slug` is a unique courier code. 
+
```

Comment:
I'm a bit confused about the relationship of `single_tracking_request` and `last_checkpoint_request`, and when you use them.  Perhaps it would be cleaner to just have people provide the full path, with a link to the API?

All trackings:

```json
{
  'path': '/tracking',
}
```

Specific:

```json
{
  'path': '/tracking'/fedex/123,
}
```

checkpoint:

```json
{
  'path': '/last_checkpoint/fedex/123'
}
```

Then the docs could say:

Provide the path for the API endpoint that you'd like to hit.  For example, for all active packages, enter `/trackings` (see https://www.aftership.com/docs/api/4/trackings), for a specific package, use `/trackings/SLUG/TRACKING_NUMBER` and replace `SLUG` with a courier code and `TRACKING_NUMBER` with the tracking number, or use Liquid (with a link).

Whatcha think?